src/Libraries/Microsoft.Extensions.Telemetry/Sampling/SamplingLoggerBuilderExtensions.cs

```diff
@@ -0,0 +1,144 @@
+using Microsoft.Extensions.DependencyInjection.Extensions;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.Shared.DiagnosticIds;
+using Microsoft.Shared.Diagnostics;
+
+namespace Microsoft.Extensions.Diagnostics.Sampling;
+
+/// <summary>
+/// Extensions for configuring logging sampling.
+/// </summary>
+[Experimental(diagnosticId: DiagnosticIds.Experiments.Telemetry, UrlFormat = DiagnosticIds.UrlFormat)]
+public static class SamplingLoggerBuilderExtensions
+{
+    /// <summary>
+    /// Adds Trace-based logging sampler to the logging infrastructure. Sampling decisions
+    /// for logs match exactly the sampling decisions for the underlying <see cref="System.Diagnostics.Activity"/>.
+    /// </summary>
+    /// <param name="builder">The dependency injection container to add logging to.</param>
+    /// <returns>The value of <paramref name="builder"/>.</returns>
+    /// <remarks>Please configure Tracing Sampling separately as part of OpenTelemetry .NET.</remarks>
+    /// <exception cref="ArgumentNullException"><paramref name="builder"/> is <see langword="null"/>.</exception>
+    public static ILoggingBuilder AddTraceBasedSampler(this ILoggingBuilder builder)
+    {
+        _ = Throw.IfNull(builder);
+
+        return builder.AddSampler<TraceBasedSampler>();
+    }
+
+    /// <summary>
+    /// Configures <see cref="RatioBasedSamplerOptions" /> from an instance of <see cref="IConfiguration" />.
+    /// </summary>
+    /// <param name="builder">The <see cref="ILoggingBuilder"/> to use.</param>
+    /// <param name="configuration">The <see cref="IConfiguration" /> to add.</param>
+    /// <returns>The value of <paramref name="builder"/>.</returns>
+    /// <exception cref="ArgumentNullException"><paramref name="builder"/> is <see langword="null"/>.</exception>
+    public static ILoggingBuilder AddRatioBasedSamplerConfiguration(this ILoggingBuilder builder, IConfiguration configuration)
+    {
+        _ = Throw.IfNull(builder);
+
+        _ = builder.Services.AddSingleton<IConfigureOptions<RatioBasedSamplerOptions>>(new RatioBasedSamplerConfigureOptions(configuration));
+
+        return builder;
+    }
+
+    /// <summary>
+    /// Adds Ratio-based sampler to the logging infrastructure. Matched logs will be sampled
+    /// according to the provided probability./>.
+    /// Higher the probability value, higher is the probability of a given log record to be sampled in.
+    /// </summary>
+    /// <param name="builder">The dependency injection container to add logging to.</param>
+    /// <param name="configuration">The <see cref="IConfiguration" /> to add.</param>
+    /// <returns>The value of <paramref name="builder"/>.</returns>
+    /// <exception cref="ArgumentNullException"><paramref name="builder"/> is <see langword="null"/>.</exception>
+    public static ILoggingBuilder AddRatioBasedSampler(this ILoggingBuilder builder, IConfiguration configuration)
```

Comment:
I'm thinking we may want to rename all the places that say 'RatioBasedSampler' to 'ProbabilitySampler'. What do other folks think?

Reply:
Renamed